packages/material/src/complex/table-array.control.tsx

```diff
@@ -41,117 +55,325 @@ export const tableArrayTester: RankedTester = rankWith(10, and(
+              </TableCell>
+              {generateCells(TableHeaderCell, resolvedSchema, path)}
             </TableRow>
           </TableHead>
           <TableBody>
-          {
-            (!data || !Array.isArray(data) || data.length === 0) ?
-              <TableRow><TableCell>No data</TableCell></TableRow> : data.map((_child, index) => {
-              const childPath = Paths.compose(path, index + '');
-
-              return (
-                <TableRow key={childPath}>
-                  {
-                    _.chain(resolvedSchema.properties)
-                      .keys()
-                      .filter(prop => resolvedSchema.properties[prop].type !== 'array')
-                      .map((prop, idx) => {
-                        return (
-                          <TableCell key={Paths.compose(childPath, idx.toString())}>
-                            <DispatchField
-                              schema={resolvedSchema}
-                              uischema={createControlElement(prop)}
-                              path={childPath}
-                            />
-                          </TableCell>
-                        );
-                      })
-                      .value()
-                  }
-                </TableRow>
-              );
-            })
-          }
+            {isEmptyTable ? <EmptyTable/> : <TableWithContent {...tableProps}/>}
           </TableBody>
         </Table>
+        <Dialog
+          open={this.state.openConfirmDelete}
+          keepMounted
+          onClose={handleClose}
+          aria-labelledby='alert-dialog-confirmdelete-title'
+          aria-describedby='alert-dialog-confirmdelete-description'
+        >
+          <DialogTitle id='alert-dialog-confirmdelete-title'>
+            {'Confirm Deletion'}
+          </DialogTitle>
+          <DialogContent>
+            <DialogContentText id='alert-dialog-confirmdelete-description'>
+              Are you sure you want to delete the {this.state.selected.length} selected objects?
+            </DialogContentText>
+          </DialogContent>
+          <DialogActions>
+            <Button onClick={handleClose} color='primary'>
+              No
+            </Button>
+            <Button onClick={removeItems} color='primary'>
+              Yes
+            </Button>
+          </DialogActions>
+        </Dialog>
       </div>
     );
   }
 }
+export interface TableProps extends ControlProps {
+  childErrors: ErrorObject[];
+}
+// FIXME copied from validation.ts
+const errorAt = instancePath => (state): ErrorObject[] => {
+  const path = instancePath + '.';
 
+  return _.filter(state.errors, (error: ErrorObject) => error.dataPath.startsWith(path));
+};
+const mapStateToTableControlProps = (state, ownProps) => {
+  const {data, errors, classNames, label, visible, enabled, id, path, inputs, required} =
+    mapStateToControlProps(state, ownProps);
+
+  const childErrors = errorAt(path)(getValidation(state));
+
+  return {
```

Comment:
I think this can be shorted with de-structuring and spread syntax:
```js
const {path, ...props } = mapStateToControlProps(state, ownProps);
const childErrors = errorAt(path)(getValidation(state));
return {
   ...props,
  path,
  childErrors
};
```